sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/zapwrapper/helper/ZapProductMessageHelper.java

```diff
@@ -58,14 +60,23 @@ public void writeProductError(ZapWrapperRuntimeException zapWrapperRuntimeExcept
         }
     }
 
-    public void writeUserMessagesWithScannedURLs(List<String> urls) {
-        for (String url : urls) {
+    public void writeUserMessagesWithDetectedURLs(List<Map<String, ApiResponse>> results) {
+        // The map looks like this: {processed=true, statusReason=OK, method=GET,
+        // reasonNotProcessed=, messageId=6, url=http://example.com, statusCode=200}
+        for (Map<String, ApiResponse> result : results) {
+            String url = result.get("url").toString();
```

Comment:
You should handle the situation when result.get("url") returns null - or can this never happen?